src/features/ProfilePage/components/AccountInfo.tsx

```diff
@@ -17,28 +22,23 @@ import { Section, Value } from '.';
 import Text from '@/components/Text';
 import { TextButton } from '@/components/Buttons';
 
-import type { UserRole } from '@/features/Authentication/authenticationApi';
-
-type Props = {
-  userRole: UserRole;
-};
-
-const AccountInfo = ({ userRole }: Props) => {
+const AccountInfo = () => {
   const { t } = useTranslation('profile');
-  const { loginName } = useAppSelector(selectUserInfo);
+  const { id, login_name: loginName, role } = useAppSelector(selectAccount);
+  const isMentor = useAppSelector(selectIsMentor);
+  const [deleteAccount, { isSuccess: isDeleteSuccess }] =
+    useDeleteAccountMutation();
+  const [logout] = useLogoutMutation();
 
-  const isMentor = userRole === 'mentor';
+  useEffect(() => {
+    if (isDeleteSuccess) logout();
+  }, [isDeleteSuccess]);
```

Comment:
I still have one slight improvement suggestion here. We can actually dispatch actions from the api-client onQueryFulfilled too!

We can remove this effect if we change the 'deleteAccount'-mutations onQueryStarted in following way.. :p

```typescript
// here we have dispatch also as property! We can trigger actions with it.
 async onQueryStarted(_, { dispatch, queryFulfilled }) {
        try {
          await queryFulfilled;
          
          // Triger the logout mutation after deleteAccount is successful
          dispatch(apiSlice.endpoints.logout.initiate());
        } catch (err) {
        ```

Reply:
Great suggestion! Now implemented! 💪

Reply:
But you did not implement this, the unwrap thing was implemented instead? I think this is more clean

Reply:
I accidentally did both 😆 Good observation! I fixed it!